cwf/gateway/fabfile.py

```diff
@@ -453,42 +423,49 @@ def _check_docker_services(ignore_list):
     sys.exit(1)
 
 
-def _start_ue_simulator():
+def _start_ue_simulator(c_test):
     """ Starts the UE Sim Service and logs into uesim.log"""
-    with cd(CWAG_ROOT + '/services/uesim/uesim'):
-        run('tmux new -d \'go run main.go -logtostderr=true -v=9 &> %s/uesim.log\'' % CWAG_ROOT)
+    with c_test.cd(CWAG_ROOT + '/services/uesim/uesim'):
+        c_test.run(
+            env={'PATH': '$PATH:/usr/local/go/bin:/home/vagrant/go/bin'},
+            command=f'tmux new -d \'go run main.go -logtostderr=true -v=9 &> {CWAG_ROOT}/uesim.log\'',
+        )
 
 
-def _start_trfserver():
+def _start_trfserver(c_trf):
     """ Starts the traffic gen server"""
-    run('nohup iperf3 -s --json -B %s > /dev/null &' % TRF_SERVER_IP)
+    trf_cmd = f'nohup iperf3 -s --json -B {TRF_SERVER_IP} > /dev/null &'
+    c_trf.sudo('apt-get install -y dtach')
```

Comment:
Why was dtach added here?

Reply:
nohup did not work, the shell was just hanging and not exiting properly. With dtach it worked. Here is some comment on this [in the docs](https://www.fabfile.org/faq.html?highlight=nohup#why-can-t-i-run-programs-in-the-background-with-it-makes-fabric-hang)